osu.Game/Screens/Tournament/Drawings.cs

```diff
@@ -0,0 +1,367 @@
+                                        },
+                                        new OsuButton()
+                                        {
+                                            RelativeSizeAxes = Axes.X,
+
+                                            Text = "Stop random",
+                                            Action = teamsContainer.StopScrolling,
+                                        },
+                                        new OsuButton()
+                                        {
+                                            RelativeSizeAxes = Axes.X,
+
+                                            Text = "Reload",
+                                            Action = reloadTeams
+                                        }
+                                    }
+                                },
+                                new FillFlowContainer()
+                                {
+                                    Anchor = Anchor.BottomCentre,
+                                    Origin = Anchor.BottomCentre,
+
+                                    RelativeSizeAxes = Axes.X,
+                                    AutoSizeAxes = Axes.Y,
+                                    Width = 0.75f,
+
+                                    Position = new Vector2(0, -5f),
+
+                                    Direction = FillDirection.Down,
+                                    Spacing = new Vector2(0, 5f),
+
+                                    Children = new Drawable[]
+                                    {
+                                        new OsuButton()
+                                        {
+                                            RelativeSizeAxes = Axes.X,
+
+                                            Text = "Reset",
+                                            Action = () => reset(false)
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            float offset = 0;
+            for (int i = 0; i < 6; i++)
+            {
+                visualiserContainer.Add(new VisualiserLine(2 * (float)Math.PI, offset, RNG.Next(10000, 12000))
+                {
+                    RelativeSizeAxes = Axes.Both,
+                });
+
+                offset += (float)Math.PI / 6f;
+            }
+
+            teamsContainer.OnSelected += t =>
+            {
+                groupsContainer.AddTeam(t.Team);
+
+                st.Text = t.Team.FullName;
+                st.FadeIn(200);
+
+                writeResults(groupsContainer.ToStringRepresentation());
+            };
+
+            teamsContainer.OnScrollStarted += () => st.FadeOut(200);
+
+            reset(true);
+        }
+
+        private void writeResults(string text)
+        {
+            lastWriteOp = writeOp;
+            writeOp = Task.Run(async () =>

```

Comment:
to avoid having to store two local references (and avoid the weird `await` call) how about this:
```
Task writeTask = () => { /* async content */ }

if (writeOp != null)
    writeOp.ContinueWith(writeTask);
else
    writeOp = Task.Run(writeTask);
```